nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/impl/GroovyScriptEngineConfigurator.java

```diff
@@ -42,13 +49,56 @@ public String getScriptEngineName() {
     }
 
     @Override
-    public Object init(ScriptEngine engine, String modulePath) throws ScriptException {
+    public URL[] getModuleURLsForClasspath(String[] modulePaths, ComponentLog log) {
+        List<URL> additionalClasspath = new LinkedList<>();
+        if (modulePaths != null) {
+            for (String modulePathString : modulePaths) {
+                File modulePath = new File(modulePathString);
+
+                if (modulePath.exists()) {
+                    // Add the URL of this path
+                    try {
+                        additionalClasspath.add(modulePath.toURI().toURL());
+                    } catch (MalformedURLException mue) {
+                        log.error("{} is not a valid file/folder, ignoring", new Object[]{modulePath.getAbsolutePath()}, mue);
+                    }
+
+                    // If the path is a directory, we need to scan for JARs and add them to the classpath
+                    if (modulePath.isDirectory()) {
+                        File[] jarFiles = modulePath.listFiles(new FilenameFilter() {
+                            @Override
+                            public boolean accept(File dir, String name) {
+                                return (name != null && name.endsWith(".jar"));
+                            }
+                        });
+
+                        if (jarFiles != null) {
+                            // Add each to the classpath
+                            for (File jarFile : jarFiles) {
+                                try {
+                                    additionalClasspath.add(jarFile.toURI().toURL());
+
+                                } catch (MalformedURLException mue) {
+                                    log.error("{} is not a valid file/folder, ignoring", new Object[]{modulePath.getAbsolutePath()}, mue);
+                                }
+                            }
+                        }
+                    }
+                }
+                log.info("{} does not exist, ignoring", new Object[]{modulePath.getAbsolutePath()});
```

Comment:
I think this is supposed to go in an 'else' :)
Also, can we change this to a WARN level instead of INFO please?